modules/platforms/cpp/thin-client/include/ignite/thin/transactions/transactions.h

```diff
@@ -0,0 +1,189 @@
+
+                /**
+                 * Assignment operator.
+                 *
+                 * @param other Another instance.
+                 * @return *this.
+                 */
+                ClientTransaction& operator=(const ClientTransaction& other)
+                {
+                    proxy = other.proxy;
+
+                    return *this;
+                }
+            private:
+                /** Implementation. */
+                TransactionProxy proxy;
+
+                /**
+                 * Default constructor.
+                 */
+                ClientTransaction() {}
+            };
+
+            /**
+             * Transactions client.
+             *
+             * This is an entry point for Thin C++ Ignite transactions.
+             *
+             * This class implemented as a reference to an implementation so copying of this class instance will only
+             * create another reference to the same underlying object. Underlying object released automatically once all
+             * the instances are destructed.
+             */
+            class ClientTransactions {
+            public:
+                /**
+                 * Constructor.
+                 *
+                 * @param impl Implementation.
+                 */
+                ClientTransactions(common::concurrent::SharedPointer<void> impl) :
+                    proxy(impl),
+                    label("")
+                {
+                    // No-op.
+                }
+
+                /**
+                 * Destructor.
+                 */
+                ~ClientTransactions() {}
+
+                /**
+                 * Start new transaction with completely clarify parameters.
+                 *
+                 * @param concurrency Transaction concurrency.
+                 * @param isolation Transaction isolation.
+                 * @param timeout Transaction timeout.
+                 * @param txSize Number of entries participating in transaction (may be approximate).
+                 *
+                 * @return ClientTransaction implementation.
+                 */
+                ClientTransaction TxStart(
+                        TransactionConcurrency::Type concurrency = TransactionConcurrency::PESSIMISTIC,
+                        TransactionIsolation::Type isolation = TransactionIsolation::READ_COMMITTED,
+                        int64_t timeout = 0,
+                        int32_t txSize = 0)
+                {
+                    return ClientTransaction(proxy.txStart(concurrency, isolation, timeout, txSize, label));
+                }
+
+                /**
+                 * Returns instance of {@code ClientTransactions} to mark each new transaction with a specified label.
+                 *
+                 * @param label Transaction label.
+                 * @return ClientTransactions implementation.
+                 */
+                ClientTransactions withLabel(const std::string& lbl)
```

Comment:
What if somebody do something like this:
```
ClientTransactions getTxWithRandomLabel(const ClientTransactions& cl_tx) {
         std::string rnd = getRandomString();
        
         return cl_tx.withLabel(rnd);
}
```
I think that something bad can happens:)